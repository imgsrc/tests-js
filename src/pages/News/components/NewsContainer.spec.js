import React from 'react'
import { shallow } from 'enzyme'
import { NewsContainer } from './NewsContainer'

describe('News container', () => {
  const props = {
    news: {
      data: [],
      isLoading: false,
      errorMsg: null
    },
    onGetNews: () => {
    }
  }

  describe('News container initial', () => {
    const mockGetNews = jest.fn()
    const nextProps = {
      ...props,
      onGetNews: mockGetNews
    }
    const newsContainer = shallow(<NewsContainer { ...nextProps } />)

    it('renders properly', () => {
      expect(newsContainer).toMatchSnapshot()
    })

    it('dispatches the `onGetNews()` method it receives from props', () => {
      expect(mockGetNews).toHaveBeenCalledTimes(1)
    })
  })

  describe('News container isLoading', () => {
    const nextProps = {
      ...props,
      news: {
        ...props.news,
        isLoading: true
      }
    }

    const newsContainer = shallow(<NewsContainer { ...nextProps }/>)
    // console.log(newsContainer.debug())
    expect(newsContainer.find('p').text()).toEqual('Loading...')
  })

  describe('News container render <NewsList />', () => {
    const nextProps = {
      ...props,
      news: {
        ...props.news,
        data: [1]
      }
    }
    const newsContainer = shallow(<NewsContainer { ...nextProps }/>)

    it('render <NewsList /> template', () => {
      expect(newsContainer.find('NewsList')).toHaveLength(1)
    })
  })

  describe('News container with errorMsg', () => {
    const nextProps = {
      ...props,
      news: {
        ...props.news,
        errorMsg: 'Something going wrong'
      }
    }

    const newsContainer = shallow(<NewsContainer { ...nextProps } />)

    it('renders properly', () => {
      expect(newsContainer).toMatchSnapshot()
    })

    it('renders errorMsg', () => {
      expect(newsContainer.find('p').text()).toEqual(nextProps.news.errorMsg)
    })
  })
})
