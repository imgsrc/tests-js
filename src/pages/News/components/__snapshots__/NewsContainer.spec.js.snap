// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News container News container initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsContainer
    news={
        Object {
            "data": Array [],
            "errorMsg": null,
            "isLoading": false,
          }
    }
    onGetNews={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        null,
        0,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      0,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          null,
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        0,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`News container News container with errorMsg renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsContainer
    news={
        Object {
            "data": Array [],
            "errorMsg": "Something going wrong",
            "isLoading": false,
          }
    }
    onGetNews={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <p>
          Something going wrong
</p>,
        null,
        0,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Something going wrong",
        },
        "ref": null,
        "rendered": "Something going wrong",
        "type": "p",
      },
      null,
      0,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <p>
            Something going wrong
</p>,
          null,
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something going wrong",
          },
          "ref": null,
          "rendered": "Something going wrong",
          "type": "p",
        },
        null,
        0,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
